File: carpartsshop/angularclient/src/app/model/carPart/car-part.ts
import {Car} from "../car/car";
import {BaseEntity} from "../BaseEntity";

export class CarPart extends BaseEntity{
  name: string | undefined;
  price: Number | undefined;
  description: string | undefined;
  stock: Number | undefined;
  car: Car | undefined;

  constructor(id: number, name: string | undefined, price: Number | undefined, description: string | undefined, stock: Number | undefined, car: Car | undefined) {
    super(id);
    this.name = name;
    this.price = price;
    this.description = description;
    this.stock = stock;
    this.car = car;
  }
}

File: carpartsshop/src/main/java/fsega/project/carpartsshop/model/CarPart.java
package fsega.project.carpartsshop.model;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class CarPart extends BaseEntity{
    private String name;
    private Double price;
    private String description;
    private Integer stock;
    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private Car car;
}
